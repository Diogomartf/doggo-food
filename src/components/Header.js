import React from 'react';
import styled from 'styled-components';

import Nav from './Nav';
import SearchBox from './SearchBox';
import RecentSearches from './RecentSearches';

const Header = ({
  handleSubmit,
  handleChange,
  searchText,
  inputFocus,
  handleOnFocus,
  handleOnBlur,
  recentSearches
}) => (
  <div>
    <Nav />
    <Relative>
      <SearchBox
        searchText={searchText}
        handleSubmit={handleSubmit}
        handleChange={handleChange}
        handleOnFocus={handleOnFocus}
        handleOnBlur={handleOnBlur}
      />
      {inputFocus && <RecentSearches searches={recentSearches} />}
    </Relative>
  </div>
);

const Relative = styled.div`
  position: relative;
`;

export default Header;
