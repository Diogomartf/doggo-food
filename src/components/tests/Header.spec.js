import React from 'react';
import { shallow } from 'enzyme';

import Header from '../Header';

describe('<Header/>', () => {
  it('should render correctly', () => {
    const searchInputMock = jest.fn();
    const handleChangeMock = jest.fn();

    const wrapper = shallow(
      <Header
        handleSubmit={searchInputMock}
        handleChange={handleChangeMock}
        searchText=""
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
