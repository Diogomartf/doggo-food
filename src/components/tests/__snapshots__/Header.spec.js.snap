// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should render correctly 1`] = `
<div>
  <Nav />
  <SearchBox
    handleChange={[MockFunction]}
    handleSubmit={[MockFunction]}
    searchText=""
  />
</div>
`;
